app/models/concerns/client_data_mixin.rb

```diff
@@ -69,8 +69,20 @@ def self.csv_headers
       column_names.sort.map { |attribute| human_attribute_name(attribute) }
     end
 
+    def self.foreign_key_to_method_map
+      @_fk_map ||= Hash[reflect_on_all_associations(:belongs_to).map { |a| [a.foreign_key, a.name] }]
+    end
+
     def csv_fields
-      self.class.column_names.sort.map { |attribute| send(attribute) }
+      field_values = []
+      self.class.column_names.sort.each do |attribute|
```

Comment:
would it be possible to throw some of this logic into a well-named method? I think the test clarifies behavior but this is still a little bit complicated for me to read and understand